x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/indexlifecycle/LifecyclePolicy.java

```diff
@@ -75,11 +71,7 @@
      *            {@link LifecyclePolicy}.
      */
     public LifecyclePolicy(LifecycleType type, String name, Map<String, Phase> phases) {
```

Comment:
> (Maybe we need it for tests, so feel free to let me know)

correct, serialization tests break here because they are all TestLifecycleType, and all of a sudden they get back TimeseriesLifecycleType after being parsed, causing inequality.

I debated modifying the "equality" check around this, but that felt like a hack

Reply:
Another option is to have two constructors, the first passes in just name and phases and calls `this(TimeseriesLifecycleType.INSTANCE, name, phases)` and the second acts like the current constructor passing in the type as well. We can probably make the second constructor package private so we one use it in tests